src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -465,11 +466,16 @@ private function validateAndDenormalize(array $types, string $currentClass, stri
             // if a value is meant to be a string, float, int or a boolean value from the serialized representation.
             // That's why we have to transform the values, if one of these non-string basic datatypes is expected.
             if (\is_string($data) && (XmlEncoder::FORMAT === $format || CsvEncoder::FORMAT === $format)) {
-                if ('' === $data && $type->isNullable() && \in_array($type->getBuiltinType(), [Type::BUILTIN_TYPE_BOOL, Type::BUILTIN_TYPE_INT, Type::BUILTIN_TYPE_FLOAT], true)) {
+                if (
+                    '' === $data &&
+                    ($isNullable ?? $type->isNullable()) &&
```

Comment:
`$isNullable` can never be set here because of `''` vs `null` comparsion on `$data`